pkgs/development/tools/build-managers/waf/default.nix

```diff
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
     sha256 = "1xbd1lmchq9pdrplkr2db28xqafyw753qbhnrfn8pxdg3inxxqvq";
   };
 
-  buildInputs = [ python ensureNewerSourcesForZipFilesHook ];
+  nativeBuildInputs = [ buildPackages.python ensureNewerSourcesForZipFilesHook ];
```

Comment:
Wouldn't this re-introduce a python 2 reference to the closure?

Reply:
Since this will cause a rebuild anyway, how about removing the top-level override and using python3 explicitly so the splicing works properly?